config.js

```diff
@@ -488,27 +420,45 @@ function testDeprecatedArgv(optimistArgv, komisFamilyNameString) {
         ];
     for (let [oldName, newName] of deprecatedArgv) {
         if (optimistArgv[oldName]) {
-            console.log(`Deprecated: --${oldName} is no longer supported, use --${newName} instead.`);
+            throw new `Deprecated: --${oldName} is no longer supported, use --${newName} instead.`;
         }
     }
     for (let komisGRUArg of familyArrayNamesGRU(komisFamilyNameString)) {
         if (optimistArgv[komisGRUArg]) {
             for (let komi of ["auto","null"]) {
                 if (optimistArgv[komisGRUArg].split(",").includes(komi)) {
-                    console.log(`Deprecated: --${komisGRUArg} /${komi}/ is no longer supported, use --${komisGRUArg} /automatic/ instead`);
+                    throw new `Deprecated: --${komisGRUArg} /${komi}/ is no longer supported, use --${komisGRUArg} /automatic/ instead`;
                 }
             }
         }
     }
     console.log("\n");
 }
 
+// exports arrays:
+function full_r_u_ArgsFromFamilyNameStrings(full_r_u_Families) {
+    let finalArray = [];
+    for (let familyNameString of full_r_u_Families) {
+        familyArrayNamesGRU(familyNameString).forEach(e => finalArray.push(e));
+    }
+    return finalArray;
+}
 
 // optimist.argv.arg(general/ranked/unranked) to exports.(r_u).arg:
 function familyArrayNamesGRU(familyNameString) {
     return ["", "ranked", "unranked"].map(e => `${familyNameString}${e}`);
 }
 
+function argObject_r_u(optimistArgv, familyNameString) {
+    const [general,ranked,unranked] = familyArrayNamesGRU(familyNameString).map(e => optimistArgv[e]);
+    if (general && !ranked && !unranked) {
```

Comment:
How do we know the difference between `!ranked` means false or undefined?

If false, the admin set it to false himself. If undefined the admin did not set this value at all.

Thus i would expect `ranked !== undefined` instead.

We should test the case where you set genral = true and ranked = false. Will it output ranked = false and unranked = true, or will it output ranked = true and unranked = true?

Reply:
yes !
this new code seems better : 1) we define all args as arg || undefined
then we test if arg === undefined
https://github.com/online-go/gtp2ogs/pull/213/files#diff-e22b50e740e102c8de5e858dbcd7753dR466